### DISCUSSION TITLE
Integration Redgtech Automação

### DISCUSSION THREAD

**Discussion by luan-nvg** [home-assistant/architecture#880] [General] [answered]
We currently own a company called [Redgtech Automação](https://redgtech.com.br/) 
The company's proposal is that you can automate your home using the domestic assistant

this board returns you a JSON with the GET HTTP method with the board data
and uses MQTT to return real-time data

We are currently registered as an Integration Skill on Google Home and Alexa
and we have our app published in the store [Apple](https://apps.apple.com/br/app/redgtech-automa%C3%A7%C3%A3o/id1565631105) and [Playstore](https://play.google.com/store/apps/details?id=com.redgtech&hl=pt_BR&gl=US)

**Proposal:**

the user will log into his home assistant and link the company as an integration, at that moment taking some information to identify this user (username or email) we will return via GET the data of all the channels that the user has access to (if there is one way to separate by card would be rewarding since a user can have several cards with X channels)
and we would make a request to turn on or off that respective channel using HTTP, if by any chance it would be necessary to use mqtt compulsorily, we would have to implement a separate mqtt for the home assistant (that is, a specific port for this connection)

Note: we are concerned about customers who do not have much programming knowledge, we want to create a Route that already comes with the standard that is necessary for the channels to be displayed on your home assistant dashboard

a basic return example that we use in the application:
[Retornos Cloud v4.2(2).pdf](https://github.com/home-assistant/architecture/files/10933248/Retornos.Cloud.v4.2.2.pdf)

By default, Google home looks like this:

```
[
  {
                            id: id,
                            type: "action.devices.types.SWITCH",
                            traits: ["action.devices.traits.OnOff"],
                            willReportState: true,
                            name: {
                                defaultNames: [deviceName],
                                name: "deviceName"',
                                nicknames: [deviceName]
                            }
      },

  {
                            id: id,
                            type: "action.devices.types.SWITCH",
                            traits: ["action.devices.traits.OnOff"],
                            willReportState: true,
                            name: {
                                defaultNames: [deviceName],
                                name: "deviceName2",
                                nicknames: [deviceName]
                            }
      }
]
```
and alexa is like this:

```
     [  
{
                    "description": "Used to trigger",
                    "displayCategories": ["SWITCH"],
                    "endpointId": id,
                    "friendlyName": "deviceName 1",
                    "manufacturerName": "Servidor"
 },

{
                    "description": Used to trigger",
                    "displayCategories": ["SWITCH"],
                    "endpointId": id,
                    "friendlyName": "deviceName 1",
                    "manufacturerName": "Servidor"
 }

]
```

that is, our server sends a pattern respecting the pattern of both companies to display the channels according to the quantity

![WhatsApp Image 2023-03-09 at 11 57 41](https://user-images.githubusercontent.com/13324964/224064949-28cf18e8-6167-41a7-ba41-c3e1b8d41845.jpeg)

![landing-page-1024x600](https://user-images.githubusercontent.com/13324964/224060565-c2a3e8a8-c416-4690-ba19-e80a9ba47d83.png)

**Comment by frenck** [comment, marked as answer]
> the user will log into his home assistant and link the company as an integration

If you want to write an integration, feel free to do so. You can find our developer documentation here:

<https://developers.home-assistant.io>

The architectural repository is not for requesting new integrations. Nor do we develop integrations for third-parties.

../Frenck

**Reply by luan-nvg** [reply]
Hello good afternoon all right? sorry for the thread, the docs said in the following snippet: "To suggest a new domain, start a discussion in the architecture repository. Be sure to show what data your proposed entity would include and how it can be controlled. Include examples of various tags ."

could you guide me to a link where registering our company and doing this particular integration to appear as "Integrators" (brand) I confess that I am a little lost and I could not find a link that would give a north through https://developers.home-assistant.io/

**Reply by frenck** [reply]
> the docs said in the following snippet

That is about our base entities and entity components, not regular integrations.

> could you guide me to a link where registering our company and doing this particular integration to appear as "Integrators"

There is no such thing. If you want to create and contribute an integration, please check out our developer documentation on how you can build one:

https://developers.home-assistant.io/docs/creating_component_index

**Reply by luan-nvg** [reply]
I need my company to be anxious here, how do i appear?


![Captura de tela de 2023-03-09 12-06-07](https://user-images.githubusercontent.com/13324964/224157740-19f14e6b-bf06-4cd0-aad1-c3d4db1b9fc1.png)

**Reply by frenck** [reply]
Hi!

By programming an integration, for more information see the links I gave above.
You have to create, build and program it yourself. We don't do that for you.

Thanks 👍 

../Frenck
